src/Shared/Components/DynamicDataTable/DynamicDataTableRow.tsx

```diff
@@ -0,0 +1,308 @@
+import { createElement, createRef, Fragment, ReactElement, RefObject, useEffect, useRef } from 'react'
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { followCursor } from 'tippy.js'
+
+import { ReactComponent as ICClose } from '@Icons/ic-close.svg'
+import { DEFAULT_SECRET_PLACEHOLDER } from '@Shared/constants'
+import { Tooltip } from '@Common/Tooltip'
+
+import { ConditionalWrap } from '@Common/Helper'
+import { ResizableTagTextArea } from '@Common/CustomTagSelector'
+import { SelectTextArea } from '../SelectTextArea'
+import {
+    getSelectPickerOptionByValue,
+    SelectPicker,
+    SelectPickerOptionType,
+    SelectPickerVariantType,
+} from '../SelectPicker'
+import { getActionButtonPosition, getRowGridTemplateColumn, rowTypeHasInputField } from './utils'
+import { DynamicDataTableRowType, DynamicDataTableRowProps, DynamicDataTableRowDataType } from './types'
+
+const conditionalWrap =
+    <K extends string>(renderer: (row: DynamicDataTableRowType<K>) => ReactElement, row: DynamicDataTableRowType<K>) =>
+    (children: JSX.Element) => {
+        if (renderer) {
+            const { props, type } = renderer(row)
+            return createElement(type, props, children)
+        }
+
+        return null
+    }
+
+export const DynamicDataTableRow = <K extends string>({
+    rows = [],
+    headers,
+    maskValue,
+    readOnly,
+    isAdditionNotAllowed,
+    isDeletionNotAllowed,
+    validationSchema = () => ({ isValid: true, errorMessages: [] }),
+    showError,
+    actionButtonConfig = null,
+    onRowEdit,
+    onRowDelete,
+    leadingCellIcon,
+    trailingCellIcon,
+    buttonCellWrapComponent,
+}: DynamicDataTableRowProps<K>) => {
+    // CONSTANTS
+    const isFirstRowEmpty = headers.every(({ key }) => !rows[0]?.data[key].value)
+    /** Boolean determining if table has rows. */
+    const hasRows = (!readOnly && !isAdditionNotAllowed) || !!rows.length
+    const disableDeleteRow = rows.length === 1 && isFirstRowEmpty
+    /** style: grid-template-columns */
+    const rowGridTemplateColumn = getRowGridTemplateColumn(
+        headers,
+        actionButtonConfig,
+        isDeletionNotAllowed || readOnly,
+    )
```

Comment:
Are we computing this in both header and row?

Reply:
yes both have different values 

headers don't have action button and delete button